controllers/awsmachine_controller.go

```diff
@@ -373,6 +379,40 @@ func (r *AWSMachineReconciler) getOrCreate(scope *scope.MachineScope, ec2svc ser
 	return instance, nil
 }
 
+func (r *AWSMachineReconciler) getAddresses(instance *infrav1.Instance, ec2svc *ec2.Service) ([]corev1.NodeAddress, error) {
+	addresses := []corev1.NodeAddress{}
+
+	enis, err := ec2svc.GetInstanceENIs(instance.ID)
```

Comment:
every instance has one or more ENIs attached regardless of if it's specified in the spec, as you can have a pre-created ENI or an ephemeral one for this instance. changed the logic to just range through the ENIs and fish out IPs and DNS names.